app/models/miq_ae_method.rb

```diff
@@ -30,6 +30,12 @@ class MiqAeMethod < ApplicationRecord
   AVAILABLE_SCOPES     = ["class", "instance"]
   validates_inclusion_of  :scope,     :in => AVAILABLE_SCOPES
 
+  # finds by name or namespace. not domain
+  # @param [nil,String] search
+  scope :name_path_search, lambda { |search|
+    search.present? ? where(arel_table[:relative_path].matches("%#{search}%")) : where({})
```

Comment:
I think this is a potential security issue?  We try to avoid interpolation and using parameter binding instead.  What would happen if the search itself had a `%` ?

Reply:
I see one use of `sanitize_sql_like` and 15 uses of the way I presented it. (grep for 'matches.*%')
Do we actually care if someone uses `%` in the phrase?

I'll look into making these changes but since we are adding in a wild character, not sure why it buys us.

Reply:
It's more of a best practices thing?  I thought `like ?` with bindings was typically the way to do it, but if matches with interpolation also works, then I'm fine with it.

> Do we actually care if someone uses % in the phrase?

`%` in the phrase is really a question to you?  Will name_path_search break if someone has a literal `%` in their namespace path?  Is that even a valid character?  It might be worth having a test case for that particular scenario to make sure it works and that this code can find those edge cases.

Reply:
1. Verified that this pattern is not susceptible to sql injection attacks. interpolation in the main sql is, but the arguments are secure
2. If searching for a string with a %, it finds it, but treats it as a wild character
3. Method names can not have a `%` in them, but I feel our conversation may be about the bigger picture, so I'm trying not to get bound down by this